packages/http/lib/auth.tsp

```diff
@@ -111,7 +111,7 @@ model ApiKeyAuth<Location extends ApiKeyLocation, Name extends string> {
  * @template Scopes The list of OAuth2 scopes, which are common for every flow from `Flows`. This list is combined with the scopes defined in specific OAuth2 flows.
  */
 @doc("")
-model OAuth2Auth<Flows extends OAuth2Flow[], Scopes extends valueof string[] = []> {
+model OAuth2Auth<Flows extends OAuth2Flow[], Scopes extends string[] = []> {
   @doc("OAuth2 authentication")
   type: AuthType.oauth2;
```

Comment:
Should `valueof string[]` be assignable to `string[]`

for example

```
model A<T extends string[]> {}
model B<T extends valueof string[]> is A<T>;
```

Reply:
Similar question Should `#{}` be assignable to `{}` or `unknown` if that the constraint on a decorator or template param

Reply:
I think based on the principle that types and values should be as distinct as possible, my opinion so far:

> Should valueof string[] be assignable to string[]

No, ideally error on T in `A<T>`.

> Should #{} be assignable to {} or unknown if that the constraint on a decorator or template param

No, would need `valueof {}` or `valueof unknown`.